knowledge_base/Environment-Variables.md

```diff
@@ -0,0 +1,394 @@
+
+## DISCORD_BOT_URL
+
+<!-- This token appears to be a holdover from the first iteration of Discobot, but is still (likely erroneously) referenced in the Snapshot Consumer. Flagged for removal. -->
+
+## DISCORD_CLIENT_ID
+
+For local testing, we use the staging Discord app/bot. The client ID can therefore be found on the [developer dashboard](https://discord.com/developers/applications/1027997517964644453/oauth2/general) or by contacting Jake or Timothee.
+
+## DISCORD_WEBHOOK_URL_DEV
+
+Connects to the #webhook-testing Discord channel on the Commond Protocol Discord server. Required to use the emit-webhook script to send Discord webhooks. More info at [Webhooks.md](./Webhooks.md).
+
+## DL_BUILD
+
+Boolean which ensures that Heroku only builds the Discobot package and related code when deploying.
+
+## ENFORCE_SESSION_KEYS
+
+Boolean feature flag for server-side enforcement of Canvas session keys; by default, `false`.
+
+Owner: Raymond Zhong.
+
+## ETH_ALCHEMY_API_KEY
+
+Used in our CI/CD and stored in GitHub repo secrets. If set, the `load-db` package script will replace production Alchemy URLs with their locally supported variants. Only needed if doing work that involves querying Ethereum.
+
+Owner: Kurtis Assad.
+
+## ETH_RPC
+
+<!-- In need of documentation; if you have information about this token, please contribute. -->
+
+## ETHERSCAN_JS_API_KEY
+
+API key for Ethereum data.
+
+## FLAG_COMMUNITY_HOMEPAGE
+
+Boolean toggle to display side-wide homepage feature for communities. Temporary flag for 2.0 work.
+
+## FLAG_PROPOSAL_TEMPLATES
+
+Boolean toggle to display side-wide sidebar proposal templates. Temporary flag for 2.0 work.
+
+## FLAG_SIDEBAR_TOGGLE
+
+Boolean toggle to display site-wide discussions homepage sidebar. Temporary flag for 2.0 work.
+
+## HEROKU_APP_NAME
+
+Automatically in Heroku; the respective app name. Required when using the DataDog PSQL integration.
+
+## IS_CI
+
+Used and defined on-the-fly in e2e scripts.
+
+Owner: Kurtis Assad.
+
+## JWT_SECRET
+
+Required in production. The JWT seed secret that is used to generate all user JWTs.
+
+## MAGIC_API_KEY
+
+Secret API key for Magic login. Contact Jake Naviasky or Graham Johnson for access.
+
+## MAGIC_DEFAULT_CHAIN
+
+Default chain for Magic login; as of 231212, we use `ethereum`.
+
+## MAGIC_PUBLISHABLE_KEY
+
+Publishable API key for Magic login; as of 231212, development uses `pk_live_EF89AABAFB87D6F4`.
+
+## MAGIC_SUPPORTED_BASES
+
+Chain bases supported for Magic login; as of 231212, we use  `substrate,ethereum`.
+
+## MIXPANEL_DEV_TOKEN
+
+Mixpanel analytics tracking token for development work. Reach out to a lead to request a token.
+
+## MIXPANEL_PROD_TOKEN
+
+Mixpanel analytics tracking token for our live production site.
+
+## NEXT_PUBLIC_RSA_PRIVATE_KEY
+
+<!-- Introduced in #2096 but never documented. If you have information about this token, please contribute. Otherwise, this entry is flagged for removal, and the referrent code for refactoring. -->
```

Comment:
Looks related to discobot login flow -- need to determine if it's deprecated.

Reply:
Ian *believes* (I don't want to misrepresent his epistemic status) this is related to server-side JWT tokens and was deprecated in favor of session keys.

Reply:
It's only used in the `decryptWithJWE` function, which is called by "authCallback" = the `/auth/callback` route. This page seems to make a call to `${DISCORD_BOT_SUCCESS_URL}/success` which seems to relate to the old version of Discobot (created in #2099). So Ian may be correct here, and we can create a follow-up ticket to remove the authCallback route + associated env vars (DISCORD_BOT_SUCCESS_URL, NEXT_PUBLIC_RSA_PRIVATE_KEY).

Reply:
Ahead of you! https://github.com/hicommonwealth/commonwealth/issues/6185

I'll update the ticket with the associated env vars for removal